zerver/lib/actions.py

```diff
@@ -3799,11 +3799,27 @@ def do_update_message(user_profile: UserProfile, message: Message, topic_name: O
     }  # type: Dict[str, Any]
     changed_messages = [message]
 
+    send_event_users = list()  # call update event for these users
+    ums = UserMessage.objects.filter(message=message.id)
+    ums_user_ids = [user.user_profile_id for user in ums]
     if message.is_stream_message():
         stream_id = message.recipient.type_id
         event['stream_name'] = Stream.objects.get(id=stream_id).name
-
-    ums = UserMessage.objects.filter(message=message.id)
+        # Get subscribers of the stream who are active now.
+        active_subs = Subscription.objects.filter(recipient__type=Recipient.STREAM,
```

Comment:
Can you use the helper function `get_active_subscriptions_for_stream_id(stream_id)` here?  It's in `zerver/lib/stream_subscription`.

And you can `get_active_subscriptions_for_stream_id(stream_id).values("user_profile_id") ` for an easy optimization (no need to bring fat Subscription records when we only care about the id).

Reply:
It might be nice to just create a new helper called `get_active_user_ids_for_stream_id` in the library, actually.